myCodes/data_base2.cpp

```diff
@@ -0,0 +1,221 @@
+#include <iostream>
+#include <vector>
+#include <map>
+#include <set>
+#include <iomanip>
+#include <sstream>
+#include <iostream>
+
+class Date
+{
+public:
+    Date(int year, int month, int day) : m_year(year),
+                                         m_month(month),
+                                         m_day(day)
+    {
+        if (month > 12 || month <= 0)
+        {
+            throw std::runtime_error("Month value is invalid: " + std::to_string(month));
+        }
+
+        if (day > 31 || day <= 0)
+        {
+            throw std::runtime_error("Day value is invalid: " + std::to_string(day));
+        }
+    }
+    int GetYear() const
+    {
+        return m_year;
+    }
+
+    int GetMonth() const
+    {
+        return m_month;
+    }
+
+    int GetDay() const
+    {
+        return m_day;
+    }
+
+    static Date ReadDate(std::istream &stream)
+    {
+        int year, month, day;
+        std::string date;
+        std::getline(stream, date, ' ');
+
+        if (IsValidDate(year, month, day, date))
+        {
+            return Date(year, month, day);
+        }
+    }
+
+private:
+    int m_year, m_month, m_day;
+    static bool IsValidDate(int &year, int &month, int &day, std::string date)
```

Comment:
1) Лучше проверять формат даты в отдельной функции или классе
2) IsValidDate назван так, что должен только проверять валидна дата или нет, а он занимается парсингом
2.1) Один и тот же метод делает две вещи (проверяет на валидность и парсит), нарушение принципа единой ответственности (solid)
3) Строки принимаем по константной ссылке
```suggestion
    static bool IsValidDate(int &year, int &month, int &day, const std::string& date)
```